src/core/tritonserver.cc

```diff
@@ -1717,13 +1705,27 @@ TRITONSERVER_ServerNew(
   lserver->SetModelRepositoryPaths(loptions->ModelRepositoryPaths());
   lserver->SetModelControlMode(loptions->ModelControlMode());
   lserver->SetStartupModels(loptions->StartupModels());
-  lserver->SetStrictModelConfigEnabled(loptions->StrictModelConfig());
+
+  bool strict_model_config = loptions->StrictModelConfig();
+  lserver->SetStrictModelConfigEnabled(strict_model_config);
+  // Note the condition is reverted due to setting name is different
+  loptions->AddBackendConfig(
+      std::string(), "auto-complete-config",
+      strict_model_config ? "false" : "true");
+
   lserver->SetRateLimiterMode(loptions->RateLimiterMode());
   lserver->SetRateLimiterResources(loptions->RateLimiterResources());
   lserver->SetPinnedMemoryPoolByteSize(loptions->PinnedMemoryPoolByteSize());
   lserver->SetCudaMemoryPoolByteSize(loptions->CudaMemoryPoolByteSize());
-  lserver->SetMinSupportedComputeCapability(
-      loptions->MinSupportedComputeCapability());
+
+  double min_compute_capability = loptions->MinSupportedComputeCapability();
+  lserver->SetMinSupportedComputeCapability(min_compute_capability);
+  loptions->AddBackendConfig(
+      std::string(), "min-compute-capability",
+      std::to_string(min_compute_capability));
+  loptions->AddBackendConfig(
+      std::string(), "backend-directory", loptions->BackendDir());
+
   lserver->SetStrictReadinessEnabled(loptions->StrictReadiness());
   lserver->SetExitTimeoutSeconds(loptions->ExitTimeout());
   lserver->SetBackendCmdlineConfig(loptions->BackendCmdlineConfigMap());
```

Comment:
This line is where all the AddBackendConfig commands must be completed before. Can you move the AddBackendConfig calls and this line does to after the other option setting (line1740) and put a comment indicating as much.

Reply:
Done